Services/Simulation/JavascriptInterpreter.cs

```diff
@@ -53,19 +53,28 @@ public class JavascriptInterpreter : IJavascriptInterpreter
             // logging into the service logs
             engine.SetValue("log", new Action<object>(this.JsLog));
 
-            //register callback for state updates
-            engine.SetValue("updateState", new Action<JsValue>(this.UpdateState));
+            // Register callback for state updates
+            engine.SetValue("updateState", new Action<JsValue>(this.JsMergeState));
+            engine.SetValue("reportMethodProgress", new Action<string>(this.JsReportMethodProgress));
```

Comment:
I'm not sure it's worth changing method progress.  We don't use the method progression reported properties in the twin from the UI right now.  Method progress itself needs redesigned and the sdk support for individual device execution of methods needs evaluated.